rules/abstract-rector/src/Rector/AbstractConvertToAnonymousFunctionRector.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\AbstractRector\Rector;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\Closure;
+use PhpParser\Node\Expr\ClosureUse;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Param;
+use PhpParser\Node\Stmt;
+use PhpParser\Node\Stmt\Expression;
+use Rector\Core\Rector\AbstractRector;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+/**
+ * @see https://www.php.net/functions.anonymous
+ */
+abstract class AbstractConvertToAnonymousFunctionRector extends AbstractRector
+{
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->shouldSkip($node)) {
+            return null;
+        }
+
+        $parameters = $this->getParameters($node);
+        $body = $this->getBody($node);
+        $useVariables = $this->resolveUseVariables($body, $parameters);
+
+        $anonymousFunctionNode = new Closure();
+
+        foreach ($parameters as $parameter) {
+            $anonymousFunctionNode->params[] = new Param($parameter);
+        }
```

Comment:
What about direct assign of all parameters?

Reply:
Right, I'll check it out

Reply:
Ok done